source/wifi/wifi_hal.c

```diff
@@ -7199,30 +7199,40 @@ INT wifi_getApAssociatedDeviceTidStatsResult(INT radioIndex,  mac_address_t *cli
 
 INT wifi_startNeighborScan(INT apIndex, wifi_neighborScanMode_t scan_mode, INT dwell_time, UINT chan_num, UINT *chan_list)
 {
-    INT status = RETURN_ERR;
-    UINT index;
     INT Radio_Index;
-    char cmd[256]={0};
+    char cmd[MAX_CMD_SIZE]={0};
     WIFI_ENTRY_EXIT_DEBUG("Inside %s:%d\n",__func__, __LINE__);
-    wifi_getApRadioIndex(apIndex,&Radio_Index);	
+    int ret = wifi_getApRadioIndex(apIndex,&Radio_Index);
+    if ( ret != 0 )
+    {
+        printf("Failed to get Radio index \n");
+        return RETURN_ERR;
+    }
     if (scan_mode == WIFI_RADIO_SCAN_MODE_ONCHAN)
-    {   
+    {
         return RETURN_OK;
     }
      /* If channels are not specified use default driver params */
     if (chan_num)
     {
         uint32_t    chan_index;
         for (chan_index = 0; chan_index < chan_num; chan_index++)
-        {   
-     	    snprintf(cmd,sizeof(cmd),  "iw dev %s%d scan -u freq %u passive >> /tmp/%u.txt ", RADIO_PREFIX, Radio_Index, chan_list[chan_index],chan_list[chan_index]);
-	    system(cmd);
+        {
+            int freq = 0;
+            if( chan_to_freq(Radio_Index,chan_list[chan_index],&freq) == RETURN_ERR )
+            {
+                printf("Failed to get frequency from the chan_to_freq");
```

Comment:
Please use wifi_dbg_printf instead of printf and add \n on the end

Reply:
updated